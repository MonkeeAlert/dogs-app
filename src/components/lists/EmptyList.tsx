import {ColorValue, StyleSheet, Text, View} from 'react-native';
import React from 'react';
import {Icon} from 'react-native-elements';
import {colors} from '../../utils/constants';

interface Props {
  title?: string;
  color?: ColorValue;
}

const EmptyList = ({title, color}: Props) => {
  return (
    <View>
      <Text style={[styles.text, {color: color ?? colors.turquoise}]}>
        {title ?? 'List is empty'}
      </Text>
      <Icon
        name={'sad-outline'}
        type={'ionicon'}
        size={28}
        color={color ?? colors.turquoise}
        tvParallaxProperties={false}
      />
    </View>
  );
};

export default EmptyList;

const styles = StyleSheet.create({
  text: {
    textAlign: 'center',
    marginBottom: 7,
    fontSize: 21,
    fontWeight: 'bold',
  },
});
