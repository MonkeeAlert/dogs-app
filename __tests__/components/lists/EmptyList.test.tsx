import React from 'react';
import {create} from 'react-test-renderer';
import EmptyList from '../../../src/components/lists/EmptyList';

describe('EmptyList', () => {
  it('matches snapshot', () => {
    const tree = create(<EmptyList />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
