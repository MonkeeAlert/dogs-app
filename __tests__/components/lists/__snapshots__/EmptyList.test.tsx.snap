// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyList matches snapshot 1`] = `
<View>
  <Text
    style={
      Array [
        Object {
          "fontSize": 21,
          "fontWeight": "bold",
          "marginBottom": 7,
          "textAlign": "center",
        },
        Object {
          "color": "#2BD9A5",
        },
      ]
    }
  >
    List is empty
  </Text>
  <View
    style={
      Object {
        "overflow": "hidden",
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "justifyContent": "center",
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#2BD9A5",
                "fontSize": 28,
              },
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          testID="iconIcon"
        >
          ï“·
        </Text>
      </View>
    </View>
  </View>
</View>
`;
